### ISSUE TITLE
Target location in Notebook Properties - accidentally deleted

### ISSUE THREAD

**Issue by sdouglas-usgs** [jupyter/notebook#1934]
I was in the Jupyter Notebook properties to change the "Start In" location for my notebook.

For some reason when I Applied my changes and tried to restart notebook, the program would give me an error. I don't have the exact error now, but when I went back into properties, I could see that the Target path was clearly cut off somehow. 

Is there anyway I can fix this? It looks like there are several files that the program 'targets' in the "...Continuum\Anaconda\" folders

**Comment by takluyver** [comment]
It sounds like you're talking about the properties of a Windows shortcut? Do you have another computer which you could compare it against?

**Comment by sdouglas-usgs** [comment]
Ah yes, sorry. This is on Windows 7 - 64 bit. I don't have one now, but I could install anaconda on a separate computer and see what the paths are.

Good idea.

**Comment by sdouglas-usgs** [comment]
Ok, so I've done this and this is the problem I had last time.

When you go into the properties, the text box must have a limit because it cuts off. Here is the list of files in the target:

C:\Users\sdouglas\AppData\Local\Continuum\Anaconda3\python.exe C:\Users\sdouglas\AppData\Local\Continuum\Anaconda3\cwp.py C:\Users\sdouglas\AppData\Local\Continuum\Anaconda3 "C:/Users/sdouglas/AppData/Local/Continuum/Anaconda3/python.exe" 
"C:/Users/sdouglas/A

As you can see the last one is cut off. I didn't do this, the properties does this and if you hit "Apply" it will cut off the path and stop working.

Is there another way to get this information and update this? Is there a way to simply reinstall jupyter notebook without reinstalling Anaconda?

**Comment by takluyver** [comment]
You could try `conda remove notebook` and `conda install notebook`. It's a bit rubbish that it gets cut off, but I don't think there's anything we can do about it - Anaconda is creating any start menu shortcuts, not us.

**Comment by sdouglas-usgs** [comment]
Thanks, I'll give that a try.

I should note, the program does still work if I simply use the command line and set my cd and run Jupyter Notebook. The above problems are from using the Notebook link saved to my desktop.

**Comment by takluyver** [comment]
Closing as I don't think there's anything for us to fix here.

**Comment by 53RT** [comment]
Had the same problem and during the removal of notebook it printed the target. So for completeness I post the content for anyone else who want to solve the problem by recovering the entry in the shortcut file.

C:\Users\USERNAME\AppData\Local\Continuum\anaconda3\python.exe, args are 

['C:\\Users\\USERNAME\\AppData\\Local\\Continuum\\anaconda3\\cwp.py',
 'C:\\Users\\USERNAME\\AppData\\Local\\Continuum\\anaconda3',
 'C:\\Users\\USERNAME\\AppData\\Local\\Continuum\\anaconda3\\python.exe',
 'C:\\Users\\USERNAME\\AppData\\Local\\Continuum\\anaconda3\\Scripts\\jupyter-notebook-script.py',
 '%USERPROFILE%']

**Comment by Pit-Storm** [comment]
While installing jupyter over conda, one should set a windows environment variable to the jupyter or conda folder (e.g. %conda% ). Then the shortcuts should be created with the windows env-variable included. This will shorten the path, so it is less than 255 chars.
I found this solution in a stack-overflow answer. If you installed jupyter/anaconda in local-app-data you can use the %LOCALAPPDATA% variable.

The target string for base/root env will look like this:
`%LOCALAPPDATA%\Continuum\Anaconda3\python.exe %LOCALAPPDATA%\Continuum\Anaconda3\cwp.py %LOCALAPPDATA%\Continuum\Anaconda3 "%LOCALAPPDATA%/Continuum/Anaconda3/python.exe" "%LOCALAPPDATA%/Continuum/Anaconda3/Scripts/jupyter-notebook-script.py"`

Its logic is:
<Path-to-conda-python.exe> <path-to-conda-cwp.py> <path-to-conda-env-folder> "<path-to-conda-env-python.exe>" "<path-to-conda-env/Scripts/jupyter-notebook-script.py>"

Hope that helps.

**Comment by Pit-Storm** [comment]
While installing jupyter over conda, one should set a windows environment variable to the jupyter or conda folder (e.g. %conda% ). Then the shortcuts should be created with the windows env-variable included. This will shorten the path, so it is less than 255 chars.
I found this solution in a stack-overflow answer. If you installed jupyter/anaconda in local-app-data you can use the %LOCALAPPDATA% variable.

The target string for base/root env will look like this:
`%LOCALAPPDATA%\Continuum\Anaconda3\python.exe %LOCALAPPDATA%\Continuum\Anaconda3\cwp.py %LOCALAPPDATA%\Continuum\Anaconda3 "%LOCALAPPDATA%/Continuum/Anaconda3/python.exe" "%LOCALAPPDATA%/Continuum/Anaconda3/Scripts/jupyter-notebook-script.py"`

Its logic is:
<Path-to-conda-python.exe> <path-to-conda-cwp.py> <path-to-conda-env-folder> "<path-to-conda-env-python.exe>" "<path-to-conda-env/Scripts/jupyter-notebook-script.py>"

Hope that helps.

**Comment by abh1shekvyas** [comment]
I also have faced the same problem:

What I did was opened the Target for another File in the same folder, in my case I copied the target for Spyder 

[C:\Users\dell\anaconda3\pythonw.exe C:\Users\dell\anaconda3\cwp.py C:\Users\dell\anaconda3 C:\Users\dell\anaconda3\pythonw.exe C:\Users\dell\anaconda3\Scripts\spyder-script.py]

And then copied that to Target for my Jupyter Notebook file, but instead of ''spyder-script.py'' to ''jupyter-notebook-script.py'' and somehow it worked.

Basically Target acts as an address and you need to put the correct address for the computer to search and open the file.
